### DISCUSSION TITLE
prisma $transaction arrays caveats

### DISCUSSION THREAD

**Discussion by ssukienn** [prisma/prisma#9936] [Q&A]
Hi. Question.

1. Should parameters for transaction array be only direct prisma client queries? What are sideeffects if they are not? I.e. some promise which runs few queries but still return prisma promise at the end?
2. Is it normal that if i.e. $transaction param array elem 1 saves something to db and array elem 2 (or later) try to query for this specific item (let's say uniquely) it is not in db and query returns null? I assume that all eventual ids used for this are pregenerated

I am aware of interactive transactions just want to know the limitations and possibilities of array approach.

**Comment by austincrim** [comment]
Hi @ssukienn 👋 , thanks for your questions!

> Should parameters for transaction array be only direct prisma client queries? What are sideeffects if they are not? I.e. some promise which runs few queries but still return prisma promise at the end?

As long as you make sure to return a `PrismaPromise`, you should not see any issues using `$transaction` in this way. If you see any unexpected behavior, filing a [bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) would be very helpful!

> Is it normal that if i.e. $transaction param array elem 1 saves something to db and array elem 2 (or later) try to query for this specific item (let's say uniquely) it is not in db and query returns null? I assume that all eventual ids used for this are pregenerated

For this specific scenario (dependent writes), we recommend using our [nested writes](https://www.prisma.io/docs/guides/performance-and-optimization/prisma-client-transactions-guide#dependent-writes) feature. You can still use the `$transaction` method ([outlined here](https://www.prisma.io/docs/guides/performance-and-optimization/prisma-client-transactions-guide#scenario-pre-computed-ids-and-the-transaction-api)), but you would need to pre-generate the IDs yourself.

Hope that was helpful, let me know if I can provide any more info!

**Reply by austincrim** [reply]
Hey @ssukienn 👋, wanted to follow up, did you get all your questions answered or can I help with anything else?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
